Classes/Validation/RecaptchaValidator.php

```
@@ -58,35 +59,46 @@ class RecaptchaValidator extends \TYPO3\CMS\Form\Domain\Validator\AbstractValida
-            $httpRequest->addPostParameter('response', $recaptchaCode)
-                        ->addPostParameter('secret', $siteSecret)
-                        ->addPostParameter('remoteip', $_SERVER['REMOTE_ADDR']);
+            if ($recaptchaCode && $siteSecret) {
+                /** @var RequestFactory $httpRequest */
+                $requestFactory = GeneralUtility::makeInstance(RequestFactory::class);
 
-            try {
-                /** @var \HTTP_Request2_Response $response */
-                $response = $httpRequest->send();
+                try {
+                    /** @var \Psr\Http\Message\ResponseInterface $response */
+                    $response = $requestFactory->request(
+                        'https://www.google.com/recaptcha/api/siteverify',
+                        'POST',
+                        [
+                            'form_params' => [
+                                'response' => $recaptchaCode,
+                                'secret' => $siteSecret,
+                                'remoteip' => $_SERVER['REMOTE_ADDR']
+                            ]
+                        ]
+                    );
 
-                if ($response->getStatus() === 200) {
-                    $recaptchaResult = json_decode($response->getBody(), TRUE);
+                    if ($response->getStatusCode() === 200) {
+                        $recaptchaResult = json_decode($response->getBody(), TRUE);
 
-                    if($recaptchaResult['success']) {
-                        //exit if success
-                       return;
+                        if ($recaptchaResult['success']) {
+                            //exit if success
+                            return;
+                        }
                     }
+                } catch (\Exception $e) {
+                    // will be not valid
```

Why is it not valid? Is it still not better to throw an error which can be caught by the error handler?